packages/ember-data/lib/adapters/rest-adapter.js

```diff
@@ -412,14 +412,14 @@ var RESTAdapter =  Adapter.extend(BuildURLMixin, {
     @param {Object} query
     @return {Promise} promise
   */
-  query: function(store, type, query) {
-    var url = this.buildURL(type.modelName, null, null, 'query', query);
+  query: function(store, type, options) {
+    var url = this.buildURL(type.modelName, null, null, 'query', options.query);
 
     if (this.sortQueryParams) {
-      query = this.sortQueryParams(query);
+      options.query = this.sortQueryParams(options.query);
     }
 
-    return this.ajax(url, 'GET', { data: query });
+    return this.ajax(url, 'GET', { data: options });
```

Comment:
I think this should say as `{ data: { ... } }`. Since the properties on `data` will get transformed into query parameters.